contracts/Content/UniqueContent.sol

```diff
@@ -0,0 +1,173 @@
+
+    /***************** Stored Variables *****************/
+    mapping(uint256 => LibAsset.UniqueAsset) uniqueAssetInfo;
+    uint256 uniqueIdsCounter;
+    
+    /** Asset Minting
+    * @dev If the royalties are valid and if the caller has the original item in their wallet, it takes the original asset, mints the unique asset and updates mappings
+    * @param _data LibAsset.UniqueAssetCreateData structure object
+    */
+    function mint(LibAsset.UniqueAssetCreateData memory _data) external override {
+        _verifyRoyalties(_data.royaltyReceivers, _data.royaltyRates);
+        require((IERC1155Upgradeable(_data.contentAddress).balanceOf(_data.to, _data.tokenId)) >= 1, "You must have the original item in your wallet");
+        // transfers the original asset to be locked in the unique content contract
+        IERC1155Upgradeable(_data.contentAddress).safeTransferFrom(_data.to, address(this), _data.tokenId, 1, "");
+        _mint(_data.to, uniqueIdsCounter);
+        
+        uniqueAssetInfo[uniqueIdsCounter].creator = _msgSender();
+        uniqueAssetInfo[uniqueIdsCounter].contentAddress = _data.contentAddress;
+        uniqueAssetInfo[uniqueIdsCounter].tokenId = _data.tokenId;
+        uniqueAssetInfo[uniqueIdsCounter].uniqueAssetUri.push(_data.uniqueAssetUri);
+        uniqueAssetInfo[uniqueIdsCounter].creatorLocked = _data.creatorLocked;
+        _setTokenRoyalties(uniqueIdsCounter, _data.royaltyReceivers, _data.royaltyRates);
+        
+        emit UniqueUriUpdated(uniqueIdsCounter, 0);
+        emit Mint(_msgSender(), _data);
+
+        // increment uniqueIdsCounter for next minting process
+        uniqueIdsCounter++;
+    }
+
+    /** Asset Burning
+    * @dev If the caller is the owner of the token and if the unique asset is not creator locked, it returns the original asset, burns the unique asset, and then deletes the unique asset's token info
+    * @param _uniqueId uint256 ID of token to burn
+    */
+    function burn(uint256 _uniqueId) external override {
+        require(ownerOf(_uniqueId) == _msgSender(), "You must be the owner of this token to burn it");
+        require(!uniqueAssetInfo[_uniqueId].creatorLocked, "Creator has disabled the burning of this token");
+        // transfers original asset back to caller
+        IERC1155Upgradeable(uniqueAssetInfo[_uniqueId].contentAddress).safeTransferFrom(address(this), _msgSender(), uniqueAssetInfo[_uniqueId].tokenId, 1, "");
+        _burn(_uniqueId);
+
+        uniqueAssetInfo[_uniqueId].creator = address(0);
+        uniqueAssetInfo[_uniqueId].contentAddress = address(0);
+        uniqueAssetInfo[_uniqueId].tokenId = 0;
+        uniqueAssetInfo[_uniqueId].version = 0;
+        uniqueAssetInfo[_uniqueId].creatorLocked = false;
+        delete uniqueAssetInfo[_uniqueId].uniqueAssetUri;
+        delete tokenRoyalties[_uniqueId];
```

Comment:
tokenRoyalties should only be accessible from the MultipleRoyalties contract. add a deleteTokenRoyalties() helper function